pkg/yang/types_builtin.go

```diff
@@ -588,17 +701,30 @@ func (n Number) Less(m Number) bool {
 		return true
 	case n.Kind != Negative && m.Kind == Negative:
 		return false
-	case n.Kind == Negative:
+	}
```

Comment:
I think we can add:

case n.Kind == Negative && m.Kind != Negative:
    return true